addons/escoria-core/game/scenes/transitions/esc_transition_player.gd

```diff
@@ -3,8 +3,11 @@ extends ColorRect
 class_name ESCTransitionPlayer
 
 # Emitted when the transition was played
-signal transition_done
+signal transition_done(transition_id)
 
+# Id of the transition. Allows keeping track of the actual transition 
+# being played or finished
+var transition_id: int = 0
```

Comment:
I was wondering if this might break, but the range of int should be waaaay bigger than the maximum number of transitions in a game. :)

Reply:
Indeed, that's a potential risk, but I think we can take it as very low. If this ever happens, an easy solution is to reset to 0 :D